### PR TITLE
[SKLT-2071] Rules Improvements 

### PR DESCRIPTION
**Pull request by Yasmeeen**
### :memo: Implementation summary
changes in rules 
1. add a message in time group form in shifts case (Assign schedule to each employee individually)
2. remove rule update in time group and add rule_id in update time group
3. add applies to column in rule

### CODE REVIEW

File: src/app/core/models/rules-interfaces.model.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ export class Rule{
   deleted_tardiness_rules?: TardinessRule [] = [];
   leave_type_id?: number;
   is_half_day?: boolean;
+  time_groups?:boolean;
```
Comment (Deena-kamal) [inline src/app/core/models/rules-interfaces.model.ts]: I think we need time_groups to be string not boolean, as its retrieved as a string

File: src/app/main/time-groups/edit-time-group/edit-time-group.component.ts [hunk 2]
```diff
@@ -119,16 +119,6 @@ export class EditTimeGroupComponent implements OnInit {
     this.timeGroup.time_group_schedule_attributes!.schedule_days_attributes != scheduleDays;
   }
 
-  public updateRule(){
```
Comment (Deena-kamal) [inline src/app/main/time-groups/edit-time-group/edit-time-group.component.ts:122]: Please remove it also form the tine group service

File: src/assets/i18n/ar.json [hunk 3]
```diff
@@ -194,5 +194,7 @@
     "tr_attendance_sheet_created_successfully":"Attendance Sheet Created Successfully*",
     "tr_excel_sheet":"Excel Sheet*",
     "tr_upload_excel_sheet":"Upload Excel Sheet*",
-    "tr_national_holidays_dates":"National Holidays Dates*"
+    "tr_national_holidays_dates":"National Holidays Dates*",
+    "tr_applies_to":"Applies to*",
```
Comment (Deena-kamal) [inline src/assets/i18n/ar.json:198]: also please add translations to the de.json file

Comment (Yasmeeen) [reply src/assets/i18n/ar.json:198]: there is no de file in finger print ar,en and fr

File: src/app/core/models/rules-interfaces.model.ts (hunk 1 above)
Comment (Yasmeeen) [reply src/app/core/models/rules-interfaces.model.ts]: done

File: src/app/main/time-groups/edit-time-group/edit-time-group.component.ts (hunk 2 above)
Comment (Yasmeeen) [reply src/app/main/time-groups/edit-time-group/edit-time-group.component.ts:122]: done

**Review by Deena-kamal — APPROVED** [review]

**Pull request merged**